### ISSUE TITLE
[aws_lb_target_group] Validation forces invalid path for TCP health check

### ISSUE THREAD

**Issue by whereisaaron** [terraform-providers/terraform-provider-aws#2978]
It is desirable to be able to configure either a TCP or HTTP/HTTPS health check based on variables as in the below example. Currently the validation forces the health check path to begin with a slash("/"), even when the protocol is TCP, preventing this use case. When the protocol is TCP, blank ("") is the only valid path for the AWS API.

```hcl
  health_check {
    healthy_threshold   = 2
    unhealthy_threshold = 2
    timeout             = 10
    port                = 8000
    path                = "${var.protocol == "TCP" ? "" : "/healthz"}"
    protocol            = "${var.protocol}"
    interval            = 30
    matcher             = "${var.protocol == "TCP" ? "" : "200-399"}"
  }
```

### Terraform Version
```
Terraform v0.11.1
+ provider.aws v1.7.0
```

### Affected Resource(s)
- aws_lb_target_group

### Terraform Configuration Files
```hcl
  health_check {
    healthy_threshold   = 2
    unhealthy_threshold = 2
    timeout             = 10
    port                = 8000
    path                = ""
    protocol            = "TCP"
    interval            = 30
    matcher             = ""
  }
```

### Expected Behavior
Expected to be able to use a blank 'path' for a TCP health check.
Or else the path validation should be left to the AWS API.

### Actual Behavior
```
Error: module.nlb.aws_lb_target_group.nlb[0]: "health_check.0.path" must begin with a '/' character: ""
```

### Steps to Reproduce
1. Create a `aws_lb_target_group` with `health_check.protocol="TCP"` and `health_check.path=""`
2. `terraform init`
3. `terraform apply`

### References
Validation was recently improved to enable HTTP/HTTPS health checks to be able to be specified for TCP Target Groups in PR #2906.

**Comment by whereisaaron** [comment]
Looks like there is a fairly trivial fix for this: PR #2980

Only check for "/" if the path is not blank.

**Comment by bflad** [comment]
This fix is in master and will be released in [v1.7.1](https://github.com/terraform-providers/terraform-provider-aws/milestone/9), which we are planning for later this week.

**Comment by bflad** [comment]
This has been released in [terraform-provider-aws version 1.7.1](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#171-january-19-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
